tests/formats/dataclass/test_mixins.py

```diff
@@ -1,16 +1,65 @@
 from typing import Iterator
+from unittest import mock
 from unittest import TestCase
 
 from lxml.etree import QName
 
 from tests.fixtures.books import BookForm
 from tests.fixtures.books import Books
+from tests.fixtures.defxmlschema.chapter02.example0210 import Product
+from tests.fixtures.defxmlschema.chapter13.chapter13 import ItemsType
+from tests.fixtures.defxmlschema.chapter13.chapter13 import ProductType
+from xsdata.formats.dataclass.mixins import ClassMeta
 from xsdata.formats.dataclass.mixins import ClassVar
 from xsdata.formats.dataclass.mixins import ModelInspect
 from xsdata.formats.dataclass.mixins import Tag
+from xsdata.utils import text
 
 
 class ModelInspectTests(TestCase):
+    @mock.patch.object(ModelInspect, "get_type_hints")
+    def test_class_meta_build_vars(self, mock_get_type_hints):
+        var = ClassVar(name="foo", qname=QName("foo", "bar"), type=int, tag=Tag.ELEMENT)
+        mock_get_type_hints.return_value = [var]
+
+        inspect = ModelInspect()
+        result = inspect.class_meta(ItemsType, None)
+        expected = ClassMeta(
+            name="ItemsType",
+            qname=QName("ItemsType"),
+            mixed=False,
+            vars={var.qname: var},
+        )
+        self.assertEqual(expected, result)
+        mock_get_type_hints.assert_called_once_with(ItemsType, None)
+
+    @mock.patch.object(ModelInspect, "get_type_hints", return_value=dict())
+    def test_class_meta_with_meta_namespace(self, mock_get_type_hints):
+        inspect = ModelInspect()
+        result = inspect.class_meta(Product, None)
+        self.assertEqual(QName("http://datypic.com/prod", "product"), result.qname)
+        mock_get_type_hints.assert_called_once_with(Product, "http://datypic.com/prod")
+
+    @mock.patch.object(ModelInspect, "get_type_hints", return_value=dict())
+    def test_class_meta_with_parent_ns(self, mock_get_type_hints):
+        inspect = ModelInspect()
+        result = inspect.class_meta(ProductType, "http://parent.ns")
+        self.assertEqual(QName("http://parent.ns", "ProductType"), str(result.qname))
+        mock_get_type_hints.assert_called_once_with(ProductType, "http://parent.ns")
+
+    @mock.patch.object(ModelInspect, "get_type_hints", return_value=dict())
+    def test_class_meta_with_no_meta_name_and_name_generator(self, *args):
+        inspect = ModelInspect(name_generator=lambda x: text.snake_case(x))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Lambda may not be necessary](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=4961169)